test/openssl/test_pkey_dsa.rb

```diff
@@ -92,19 +92,23 @@ def test_sign_verify_raw
     sig = key.syssign(digest)
     assert_equal true, key.sysverify(digest, sig)
     assert_equal false, key.sysverify(digest, invalid_sig)
-    assert_raise(OpenSSL::PKey::DSAError) { key.sysverify(digest, malformed_sig) }
+    assert_equal false, key.sysverify(digest, malformed_sig) if aws_lc?
+    assert_raise(OpenSSL::PKey::DSAError) { key.sysverify(digest, malformed_sig) } if !aws_lc?
```

Comment:
Could you use if-then-else instead?

Alternatively, we can add a helper method and just ignore the difference, since ruby/openssl doesn't really have to distinguish an incorrect and a corrupted signature:

```ruby
def test_aa
  assert_false_or_error { key.sysverify(digest, malformed_sig) }
end

private

def assert_false_or_error
  assert_equal(false, yield)
rescue => e
  assert_kind_of(OpenSSL::PKey::DSAError, e)
end
```